Fix webpack overriding && Add an example with local file dependencies

ndelangen opened the pull request:
Issue: -can't find it-

## What I did
Fixed a few prepublish stuff
Fixed the babelconfig so it loads presets from included 'node_modules'

## How to test
- `npm install`
- `cd examples/test-cra`
- `npm run storybook`

This should result in a successful storybook being launched, and no messages in the terminal about presets not being loaded.

ndelangen committed:
FIX webpack overriding && CLEANUP && ADD example
for testing a non-linked example based on create-react-app

xavcz commented on examples/test-cra/package.json:
```diff
@@ -0,0 +1,26 @@
+  "version": "0.1.0",
+  "private": true,
+  "devDependencies": {
+    "@kadira/storybook": "file:../../packages/react-storybook",
```
so cool to have direct dependencies on the file system! 💯

ndelangen replied:
This is to be able to do an actual npm install without linking via lerna.
So when everything is installed flat things should work as expected.

xavcz commented on examples/test-cra/package.json:
```diff
@@ -0,0 +1,26 @@
+  },
+  "dependencies": {
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2"
```
should we be on `^15.5.4` as the latest version published?

ndelangen replied:
Yeah for some reason `npm-check-updates` is no longer helping me with this 😞

xavxyz reviewed (approved):
🎉  :tada:

ndelangen committed:
FIX an issue where a test in a package was using async-await
but was not transpiring them correctly.

MOVE jest cache directory to subdirectory `.cache`

ndelangen committed:
FIX unneeded require

ndelangen committed:
MOVE all babel-dev-dependencies

ndelangen committed:
FIX republish script so it exits with !0 if babel fails

ndelangen committed:
FIX problem with duplicate package names

ndelangen committed:
CHANGE travis to cache the .cache folder
